### PR TITLE
Governance Dashboard

### PR DESCRIPTION
**Pull request by wise4rmgodroot**
## Title
* Governance Voting Dashboard

## Description

* Added a new content under the tutorial section.
* Added images under the `static/img` folder

## Screenshots/GIFs

* N/A

## Testing

* Used the `yarn build`command.

## Checklist

- [x] I have read and understood the contributing guidelines.
- [x] I have followed the style guide and formatting guidelines.
- [x] I have added appropriate comments to explain the changes.
- [x] I have tested my changes thoroughly.

## Refs

* N/A.

### CODE REVIEW

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
+---
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]: Move the governance dashboard to the Runes folder, it's part of the Runes series

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+sidebar_position: 300
+title: Create a Governance Voting Dashboard
+sidebar_label: Create a Governance Voting Dashboard
+tags: [Governance Voting Dashboard, rootstock, tutorials, rsk, ERC20, blockchain, defi]
+description: "This project is a Governance Voting Dashboard built on the Rootstock blockchain. It allows users to create and vote for teams using governance tokens (ERC20 tokens). "
+---
+
+This project is a **Governance Voting Dashboard** built on the **Rootstock blockchain**. It allows users to create and vote for teams using governance tokens [ERC20 tokens](http://localhost:3000/concepts/glossary/#e).
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]: Kindly take out the localhost link

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+description: "This project is a Governance Voting Dashboard built on the Rootstock blockchain. It allows users to create and vote for teams using governance tokens (ERC20 tokens). "
+---
+
+This project is a **Governance Voting Dashboard** built on the **Rootstock blockchain**. It allows users to create and vote for teams using governance tokens [ERC20 tokens](http://localhost:3000/concepts/glossary/#e). 
+
+The voting is managed by a smart contract, making it decentralized and transparent. Each team has a unique name, a governance token, and a leader, and users can vote by transferring tokens to their preferred teams.
+
+The dashboard is designed for token-based voting, where votes are reflected in a team's score. Higher scores indicate more community support for that team.
+
+## Features Overview
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
## Features
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+      _readyToVote = false;
+      _teamLeaders[_teams[_teamNames[i]].teamLeaderAddress] = "";
+      _teams[_teamNames[i]] = TeamInfo("", "", "", address(0), address(0), 0);
+    }
+    _teamNames = [""];
+  }
+  ```
+:::
+
+## Prerequisites
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]: Move prerequisites section before Features

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+```
+npm install
+```
+  </Step>
+  <Step title="Set Up Environment Variables">
+   
+
+Create a .env file in the project root directory and populate it with the required environment variables as listed above.
+
+#### **Required Environment Variables**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
### Environment Variables
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+
+Set the following environment variables in your .env file to ensure the project functions correctly:
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
Set the following environment variables in your `.env` file to ensure the project functions correctly:
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+
+#### **1. NEXT_PUBLIC_TEAM_MANAGER_ADDRESS**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
#### 1. NEXT_PUBLIC_TEAM_MANAGER_ADDRESS
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+
+This is the address of the TeamsManager smart contract that will be deployed to the Rootstock network. It manages the voting and team data.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
This is the address of the `TeamsManager` smart contract that will be deployed to the Rootstock network. It manages the voting and team data.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 9]
```diff
@@ -0,0 +1,370 @@
...
+
+<figure>
+<img src="/img/resources/tutorials/governance-dashaboard/NEXT_PUBLIC_TEAM_MANAGER_ADDRESS.png" alt="NEXT_PUBLIC_TEAM_MANAGER_ADDRESS"/>
+  <figcaption>NEXT_PUBLIC_TEAM_MANAGER_ADDRESS (fig 1.)</figcaption>
+</figure>
+
+:::note
+
+If you’re new to deploying contracts on Remix, check out this [tutorial](https://dev.rootstock.io/developers/quickstart/remix/) for more detailed steps.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
See the guide on how to [deploy smart contracts on Rootstock using Remix](https://dev.rootstock.io/developers/quickstart/remix/) for more detailed steps.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 10]
```diff
@@ -0,0 +1,370 @@
...
+
+:::
+
+* **How to get it**:
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
* **How to get the contract address**:  
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 11]
```diff
@@ -0,0 +1,370 @@
...
+NEXT_PUBLIC_TEAM_MANAGER_ADDRESS=<Your_Team_Manager_Contract_Address>
+```
+
+#### **2. NEXT_PUBLIC_RPC_URL**
+
+This is the RPC (Remote Procedure Call) URL that allows an application to interact with the Rootstock blockchain. An RPC URL is essentially a link to the blockchain network.
+
+:::tip[Tip]
+
+If you’re unsure how to navigate the Rootstock RPC Dashboard, refer to the [Getting Started with Rootstock RPC API](https://dev.rootstock.io/developers/rpc-api/rootstock/setup/).
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
Refer to the [Getting Started with Rootstock RPC API](https://dev.rootstock.io/developers/rpc-api/rootstock/setup/) for more information.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 12]
```diff
@@ -0,0 +1,370 @@
...
+
+:::
+
+* **How to get it**:
+
+    * Go to the [Rootstock RPC Dashboard](https://dashboard.rpc.rootstock.io/login) and create an account or log in.  
+    * Once logged in, create a new project or access an existing one to generate a testnet RPC URL.  
+    * Copy the URL and add it to your .env file like so:
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
    * Copy the URL and add it to your .env file, see example below:
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 13]
```diff
@@ -0,0 +1,370 @@
...
+* **How to get it**:  
+  * Deploy the TeamsManager.sol contract to the Rootstock network using RemixIDE, a popular tool for writing and deploying smart contracts.   
+  * Once deployed, you’ll find the contract’s address in the Remix’s console. Copy this address.  
+  * Paste this address into your .env file as shown below:
+
+```
+NEXT_PUBLIC_TEAM_MANAGER_ADDRESS=<Your_Team_Manager_Contract_Address>
+```
+
+#### **2. NEXT_PUBLIC_RPC_URL**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
#### 2. NEXT_PUBLIC_RPC_URL
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 14]
```diff
@@ -0,0 +1,370 @@
...
+
+    * Go to the [Rootstock RPC Dashboard](https://dashboard.rpc.rootstock.io/login) and create an account or log in.  
+    * Once logged in, create a new project or access an existing one to generate a testnet RPC URL.  
+    * Copy the URL and add it to your .env file like so:
+
+```
+NEXT_PUBLIC_RPC_URL=<Your_Rootstock_RPC_URL>
+```
+
+#### **3. NEXT_PUBLIC_EXPLORER**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
#### 3. NEXT_PUBLIC_EXPLORER
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 15]
```diff
@@ -0,0 +1,370 @@
...
+
+This is the blockchain explorer URL, which lets you view transaction details on the Rootstock network.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
This is the blockchain explorer URL, which allows for viewing transaction details on the Rootstock network.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 16]
```diff
@@ -0,0 +1,370 @@
...
+
+* **How to get it**:  
+  * For testnet (for testing purposes), use this URL: https://explorer.testnet.rootstock.io.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
  * Rootstock [Testnet](https://explorer.testnet.rootstock.io)  
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 17]
```diff
@@ -0,0 +1,370 @@
...
+NEXT_PUBLIC_RPC_URL=<Your_Rootstock_RPC_URL>
+```
+
+#### **3. NEXT_PUBLIC_EXPLORER**
+
+This is the blockchain explorer URL, which lets you view transaction details on the Rootstock network.
+
+* **How to get it**:  
+  * For testnet (for testing purposes), use this URL: https://explorer.testnet.rootstock.io.  
+  * For mainnet (for production), use this URL: https://explorer.rootstock.io.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
  * Rootstock [Mainnet](https://explorer.rootstock.io)
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 18]
```diff
@@ -0,0 +1,370 @@
...
+
+**Example**:
+
+```
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
```text
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 19]
```diff
@@ -0,0 +1,370 @@
...
+NEXT_PUBLIC_EXPLORER=https://explorer.testnet.rootstock.io
+```
+
+ This variable allows your app to link to transaction history directly on the explorer, providing a user-friendly way to verify actions on the blockchain.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
This variable enables your app to directly link to transaction details on the blockchain explorer, offering a seamless user experience for verifying blockchain activity.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 20]
```diff
@@ -0,0 +1,370 @@
...
+
+#### **4. NEXT_PUBLIC_PINATA_URL**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
#### 4. NEXT_PUBLIC_PINATA_URL
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 21]
```diff
@@ -0,0 +1,370 @@
...
+
+This is the URL to Pinata, an IPFS (InterPlanetary File System) provider, which will be used to store and retrieve metadata related to teams and votes.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
Here, you can find the URL to Pinata - an IPFS (InterPlanetary File System) provider, which will be used to store and retrieve metadata related to teams and votes.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 22]
```diff
@@ -0,0 +1,370 @@
...
+
+<figure>
+<img src="/img/resources/tutorials/governance-dashaboard/NEXT_PUBLIC_PINATA_URL.png" alt="NEXT_PUBLIC_PINATA_URL"/>
+  <figcaption>NEXT_PUBLIC_PINATA_URL (fig 2.)</figcaption>
+</figure>
+
+* **How to get it**:
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
* **How to get the URL**:  
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 23]
```diff
@@ -0,0 +1,370 @@
...
+
+This is the URL to Pinata, an IPFS (InterPlanetary File System) provider, which will be used to store and retrieve metadata related to teams and votes.
+
+<figure>
+<img src="/img/resources/tutorials/governance-dashaboard/NEXT_PUBLIC_PINATA_URL.png" alt="NEXT_PUBLIC_PINATA_URL"/>
+  <figcaption>NEXT_PUBLIC_PINATA_URL (fig 2.)</figcaption>
+</figure>
+
+* **How to get it**:  
+  * Start by logging into your Pinata Cloud account and navigate to the dashboard [Pinata](https://www.pinata.cloud/).
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
  * Log into your Pinata Cloud account and navigate to the [dashboard](https://www.pinata.cloud/).  
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 24]
```diff
@@ -0,0 +1,370 @@
...
+
+For example:
+
+```
+https://emerald-familiar-cobra-431.mypinata.cloud/ipfs/QmVsBy3vo9tmbjYmFNCiRwAvCBU7p9peEEgyR9awJg4Tyj
+```
+
+  * The base URL is the part of the file URL before the unique CID (Content Identifier). In this example, the base URL is:
+
+```
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]: Remove url from code block, use a link instead. Same with above and below example

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 25]
```diff
@@ -0,0 +1,370 @@
...
+https://emerald-familiar-cobra-431.mypinata.cloud/ipfs/
+```
+
+  * To display images or files from Pinata Cloud on your frontend, simply append the CID to the base URL.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
  * To display the images or files from Pinata Cloud on the frontend, simply append the CID to the base URL.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 26]
```diff
@@ -0,0 +1,370 @@
...
+
+**Example:**
+
+```
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
```text
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 27]
```diff
@@ -0,0 +1,370 @@
...
+NEXT_PUBLIC_PINATA_URL=<Your_Pinata_URL>
+```
+:::note
+
+IPFS helps in securely storing data in a decentralized way, and Pinata simplifies accessing this data on IPFS in your dApp.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
IPFS helps to secure store data in a decentralized way, and Pinata simplifies access to this data on IPFS in your dApp.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 28]
```diff
@@ -0,0 +1,370 @@
...
+
+:::
+
+#### **5. NEXT_PUBLIC_GOVERNANCE_TOKEN**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
#### 5. NEXT_PUBLIC_GOVERNANCE_TOKEN
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 29]
```diff
@@ -0,0 +1,370 @@
...
+
+ This is the address of the ERC20 governance token that users will use to vote. ERC20 tokens are a standard token type on Ethereum-compatible networks.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
This is the address of the ERC20 governance token that users will use to vote. ERC20 tokens are a standard token type on Ethereum-compatible networks.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 30]
```diff
@@ -0,0 +1,370 @@
...
+
+* #### **How to get it:**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
* #### Get the governance token address
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 31]
```diff
@@ -0,0 +1,370 @@
...
+
+  *  If you already have a governance token contract, you can deploy it on the Rootstock network, similar to the TeamsManager contract.
+
+  *  Otherwise, you can use an existing ERC20 token address if one has been provided for the governance purposes.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
  *  Alternatively, if an existing ERC20 token address has been provided for governance purposes, you can use that directly.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 32]
```diff
@@ -0,0 +1,370 @@
...
+NEXT_PUBLIC_EXPLORER=https://explorer.testnet.rootstock.io
+NEXT_PUBLIC_PINATA_URL=<Your_Pinata_URL>
+NEXT_PUBLIC_GOVERNANCE_TOKEN=<Your_Governance_Token_Address>
+```
+  </Step>
+
+<Step title="Start the Development Server">
+   Launch the development server with:
+
+```
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
```js
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 33]
```diff
@@ -0,0 +1,370 @@
...
+
+  *  Otherwise, you can use an existing ERC20 token address if one has been provided for the governance purposes.
+
+#### **Example:**
+
+```
+NEXT_PUBLIC_GOVERNANCE_TOKEN=<Your_Governance_Token_Address>
+```
+
+#### **Complete environment:**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
#### View the complete environment
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 34]
```diff
@@ -0,0 +1,370 @@
...
+
+ This is the address of the ERC20 governance token that users will use to vote. ERC20 tokens are a standard token type on Ethereum-compatible networks.
+
+* #### **How to get it:**
+
+  *  If you already have a governance token contract, you can deploy it on the Rootstock network, similar to the TeamsManager contract.
+
+  *  Otherwise, you can use an existing ERC20 token address if one has been provided for the governance purposes.
+
+#### **Example:**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
Example
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 35]
```diff
@@ -0,0 +1,370 @@
...
+
+```
+NEXT_PUBLIC_GOVERNANCE_TOKEN=<Your_Governance_Token_Address>
+```
+
+#### **Complete environment:**
+
+```
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
```text
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 36]
```diff
@@ -0,0 +1,370 @@
...
+<img src="/img/resources/tutorials/governance-dashaboard/Development-Server.png" alt="development server "/>
+  <figcaption>Development server  (fig 3.)</figcaption>
+</figure>
+  </Step>
+  
+</Steps>
+
+
+
+## **Interacting with the Frontend**
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
## Interacting with the Frontend
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 37]
```diff
@@ -0,0 +1,370 @@
...
+
+<Steps>
+<Step title="Connect Your Wallet">
+
+  * Click the **Connect Wallet** button in the top-right corner.  
+  * Choose your Web3 wallet `(e.g., MetaMask)` and connect it to the Rootstock network.  
+  * Make sure your wallet is funded with some governance for voting and RBTC tokens to cover transaction fees.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
  * Make sure your wallet is funded with some governance tokens for voting and RBTC tokens to cover transaction fees.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 38]
```diff
@@ -0,0 +1,370 @@
...
+
+</Step>
+<Step title="Explore the Teams">
+ <figure>
+  <img src="/img/resources/tutorials/governance-dashaboard/Explore-the-Teams.png" alt="Explore the Teams"/>
+  <figcaption>Explore the Teams (fig 4.)</figcaption>
+ </figure>
+
+ * Browse through the list of teams in the **Teams List** section.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]:
```suggestion
 * Explore the list of teams in the **Teams List** section.
```

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 39]
```diff
@@ -0,0 +1,370 @@
...
+</Step>
+   <Step title="Vote for a Team">
+    <figure>
+<img src="/img/resources/tutorials/governance-dashaboard/Vote-Team.png" alt="Vote for a Team"/>
+  <figcaption>Vote for a Team (fig 5.)</figcaption>
+</figure>
+
+ * To vote, find the team you want to support and click the **Vote** button in the **Option** column.  
+ * You will see a pop up screen with a field to add the amount to vote and an Add vote button.  
+ * Confirm the transaction in your wallet. You’ll need `governance tokens` to cast your vote and a small amount of `RBTC tokens` for transaction fees.
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]: Add inpage link to how to get the governance token

File: docs/04-resources/04-tutorials/governance-dashboard.md [hunk 40]
```diff
@@ -0,0 +1,370 @@
...
+   * **Meme Token Address**: Address of the meme token associated with the team.  
+   * **Team Leader Address**: Wallet address of the team leader.  
+ * Once added, the new team will appear in the Teams List, and users can start voting for it.
+
+  </Step>
+</Steps>
+
+
+
+## Conclusion
```
Comment (owans) [inline docs/04-resources/04-tutorials/governance-dashboard.md]: This is not a conclusion, kindly reword

File: docs/04-resources/04-tutorials/governance-dashboard.md (hunk 39 above)
Comment (wise4rmgodroot) [reply docs/04-resources/04-tutorials/governance-dashboard.md]: We dont have this on this doc, do you want me to reference external content?

Comment (owans) [reply docs/04-resources/04-tutorials/governance-dashboard.md]: If its available yes, how is the governance token generated then that is used for the voting?

Comment (wise4rmgodroot) [reply docs/04-resources/04-tutorials/governance-dashboard.md]: Its long process that in involves creating your erc20 then mint tokens to it. (But its something we can write about, as a new long form article )

This is a sample https://docs.alchemy.com/docs/how-to-create-a-dao-governance-token

File: docs/04-resources/06-guides/runes-rootstock/governance-dashboard.md [hunk 41]
```diff
@@ -0,0 +1,374 @@
+---
+sidebar_position: 300
+title: Create a Governance Voting Dashboard
```
Comment (owans) [inline docs/04-resources/06-guides/runes-rootstock/governance-dashboard.md]:
```suggestion
title: Build and Deploy a Governance Voting Dashboard on Rootstock
```

**Review by owans — APPROVED** [review]
LGTM!

**Pull request merged**